### PR TITLE
[KnownBits] Add KnownBits::add and KnownBits::sub helper wrappers.

### PR DESCRIPTION
**Pull request by RKSimon**

### CODE REVIEW

File: llvm/include/llvm/Support/KnownBits.h [hunk 1]
```diff
@@ -329,6 +329,19 @@ struct KnownBits {
   static KnownBits computeForSubBorrow(const KnownBits &LHS, KnownBits RHS,
                                        const KnownBits &Borrow);
 
+  /// Compute knownbits resulting from addition of LHS and RHS.
+  /// NSW and NUW flags are assumed to be false.
+  static KnownBits add(const KnownBits& LHS, const KnownBits& RHS) {
```
Comment (jayfoad) [inline llvm/include/llvm/Support/KnownBits.h:334]: Why not call it `operator+`?

Comment (RKSimon) [reply llvm/include/llvm/Support/KnownBits.h:334]: Mainly because I loathe operators for this kind of stuff :)

Comment (jayfoad) [reply llvm/include/llvm/Support/KnownBits.h:334]: Well we already use them for `&`, `|` and `^`...

Comment (goldsteinn) [reply llvm/include/llvm/Support/KnownBits.h:334]: \*cough\* and we shouldn't \*cough\*

**Comment by goldsteinn** [comment]
I'm opposed to no option for `NUW`/`NSW`. I think this will probably become the defacto API and think this basically encourages people, especially new contributors, to forgetting/ignoring the flags.

Can you add them as default false params?

**Comment by RKSimon** [comment]
I've added NSW/NUW arguments that default to false - although this means we can't easily use these in llvm::function_ref callbacks anymore

**Comment by goldsteinn** [comment]
> I've added NSW/NUW arguments that default to false - although this means we can't easily use these in llvm::function_ref callbacks anymore

You could add a wrapper `addNoWrap` and `subNoWrap` for the callback case.

**Comment by goldsteinn** [comment]
LGTM

**Review by goldsteinn — APPROVED** [review]

**Review by arsenm — APPROVED** [review]

**Pull request merged**